packages/react-form-state/src/tests/FormState.test.tsx

```diff
@@ -450,6 +450,32 @@ describe('<FormState />', () => {
       const {errors} = lastCallArgs(renderPropSpy);
       expect(errors).toEqual([...submitErrors, ...externalErrors]);
     });
+
+    it('does not reset fields if externalErrors are not provided', () => {
```

Comment:
I'm a bit stuck with this test. It actually passes with the old equality check and I'm not sure why.